extra/src/main/java/aima/extra/probability/domain/FiniteBooleanDomain.java

```diff
@@ -0,0 +1,66 @@
+package aima.extra.probability.domain;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+/**
+ * Singleton implementation of boolean domain whose values are {true, false}.
+ * 
+ * @author Ciaran O'Reilly
+ * @author Nagaraj Poti
+ */
+public class FiniteBooleanDomain extends AbstractFiniteDomain {
+
+	// PRIVATE
+
+	/**
+	 * Single instance of FiniteBooleanDomain.
+	 */
+	private static FiniteBooleanDomain onlyInstance = null;
+
+	/**
+	 * Set of possible values for boolean domain consists of {true, false}.
+	 * Consistent order is ensured.
+	 */
+	private Set<Boolean> possibleValues = new LinkedHashSet<Boolean>(Arrays.asList(Boolean.TRUE, Boolean.FALSE));
+
```

Comment:
possibleValues in the case can be static as it will never change. Additionally, if you use:

Collections.unmodifiableSet(new Linked)

it saves you instantiation an unmodifiable set wrapper every time you call getPossibleValues(). Ideaily, this pattern should be followed where applicable in all the domain implementations as it is more efficiet.